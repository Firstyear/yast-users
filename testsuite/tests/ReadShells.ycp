/**
 * File:
 *  shells.ycp
 *
 * Module:
 *  Users
 *
 * Summary:
 *  Available shells listing test.
 *
 * Authors:
 *  Petr Blahos <pblahos@suse.cz>
 *
 * $Id$
 *
 */

{
    import "Testsuite";

    import "Users";

    map READ = $[
	"target" : $[
	    "string"	:
"/bin/ash
/bin/bash
/bin/bash1
/bin/csh
/bin/false
/bin/ksh
/bin/sh",
	    "stat"	: $[ 1: 1], // means 'file exists'
	    "tmpdir"	: "/tmp/YaST",
	]
    ];

    Testsuite::Test (``(Users::ReadAllShells ()), [READ, $[], $[]], 0);
    Testsuite::Dump ("==========================================================");
    Testsuite::Dump (sformat ("shells: %1", Users::AllShells()));
}
