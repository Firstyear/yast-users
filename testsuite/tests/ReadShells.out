Read	.target.string "/etc/shells" "/bin/ash\n/bin/bash\n/bin/bash1\n/bin/csh\n/bin/false\n/bin/ksh\n/bin/sh"
Read	.target.size "/bin/ash" 0
Read	.target.size "/bin/bash" 0
Read	.target.size "/bin/csh" 0
Read	.target.size "/bin/false" 0
Read	.target.size "/bin/ksh" 0
Read	.target.size "/bin/sh" 0
Return	nil
Dump	==========================================================
Dump	shells: ["/bin/ash", "/bin/bash", "/bin/csh", "/bin/false", "/bin/ksh", "/bin/sh"]
